cronjobs/add_authority_external_ref.py

```diff
@@ -0,0 +1,178 @@
+        newer_gz_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_int) + ".jsonld.gz"
+        newer_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_int) + ".jsonld"
+        print("Process 2/7 -- Downloading file")
+        if DownloadTheFile(newer_gz_file_name):
+            # b. The download file is a zip file, it needs to be extracted first
+            print("Process 3/7 -- Extracting the new file (.gz)")
+            util.ExecOrDie(util.Which("gunzip"), [
+                           "-f", f"/tmp/{newer_gz_file_name}"])
+            #  c. Get the information needed by gnd_wiki and put it into the file
+            print(
+                "Process 4/7 -- Parse the file and extract the essential information needed")
+            jq_prog_with_pipe = f"jq -c --stream '.' < /tmp/{newer_file_name} | grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > /tmp/{input_file_name_for_add_auth}"
+            if os.system(jq_prog_with_pipe) == 0:
+                # d. Run "add_authority_external_ref" program
+                print("Process 5/7 -- Generate a new gnd_wiki_file")
+                util.ExecOrDie(util.Which("add_authority_external_ref"), [
+                               "--create_mapping_file", f"/tmp/{input_file_name_for_add_auth}", f"/tmp/{gnd_wiki_file}"])
+
+                print("Process 6/7 -- Copy the gnd_wiki_file to its destination")
+                util.ExecOrDie(util.Which("cp"), [
+                               "-f", f"/tmp/{gnd_wiki_file}", f"{share_folder}{gnd_wiki_file}"])
+
+                #  e. Update the latest version date on the config file (the second line)
+                print(
+                    "Process 7/7 -- Updating the config file and remove temporary files")
+                UpdateConfigFile(
+                    config_file, current_file_date_int)
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{newer_file_name}"])
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{input_file_name_for_add_auth}"])
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{gnd_wiki_file}"])
+
+                util.SendEmail("gnd_wiki_file generator",
+                               "Successfully generated gnd_wiki_file.", priority=5)
+
+            else:
+                util.SendEmail("gnd_wiki_file generator",
+                               "Error while running: " + jq_prog_with_pipe, priority=1)
+        else:
+            util.SendEmail("gnd_wiki_file generator",
+                           "Successfully generated gnd_wiki_file.", priority=1)
+    else:
+        util.SendEmail("gnd_wiki_file generator",
+                       "The file version on the web is the same or older.", priority=5)
+
+
+try:
+    Main()
+except Exception as e:
+    util.SendEmail("gnd_wiki_file generator", "An unexpected error occurred: "
```

Comment:
gnd_wiki_file generator => maybe we should replace the subject of the email by: `os.path.basename(__file__)`